beacon-chain/forkchoice/doubly-linked-tree/store.go

```diff
@@ -173,6 +173,15 @@ func (s *Store) insert(ctx context.Context,
 	processedBlockCount.Inc()
 	nodeCount.Set(float64(len(s.nodeByRoot)))
 
+	// Only update received block slot if it's within epoch from the highest received slot.
+	if slot+params.BeaconConfig().SlotsPerEpoch > s.highestReceivedSlot {
```

Comment:
why this condition instead of using the current wall clock slot?

Reply:
Good point. Done